### ISSUE TITLE
Scheduled Content: Update created_by/updated_by when post is published

### ISSUE THREAD

**Issue by andwei** [humhub/humhub#6665]
### Is your feature request related to a problem? Please describe.
The way the "Schedule publication" feature works right now is problematic for us. We mainly use it to plan announcements/reminders, sometimes weeks before the actual event happens.

We had several users complain/reach out because they were confused by the "old" date of the post, because as it is now, the schedule feature uses the original creation time. This has made this feature unusable for us, so that we are back to manually publishing the posts at the correct time, which is very unsatisfactory.

### Describe the solution you'd like
An additional checkbox in the Scheduling Options dialog "Update creation time" (or something like that). If this checkbox is set, the `created_at`/`updated_at` fields for that post/content should be updated to the time it was scheduled (or alternatively to the `scheduled_at` time).

Alternatively, if this would be considered a bug, it would suffice (for us) to actually always create scheduled posts with a created_at timestamp that corresponds to its scheduled_at timestamp (and basically the same for drafts, I do not think it makes much sense to use to original creation time of the draft as the timestamp of the post that is created from it). However, as this changes the current behavior, it may be a not-so-great-idea.

### Describe alternatives you've considered
none

### Suggestion
I've played around with the code and managed to (always, no option) get the behavior I wanted by adding:
```diff
--- a/protected/humhub/modules/content/services/ContentStateService.php
+++ b/protected/humhub/modules/content/services/ContentStateService.php
@@ -7,9 +7,12 @@
 
 namespace humhub\modules\content\services;
 
+use DateTime;
+use DateTimeZone;
 use humhub\libs\DbDateValidator;
 use humhub\modules\content\models\Content;
 use yii\base\Component;
+use Yii;
 
 /**
  * This service is used to extend Content record for state features
@@ -141,6 +144,14 @@ class ContentStateService extends Component
             }
         }
 
+        if ($state === Content::STATE_PUBLISHED) {
+          if (!empty($this->content->scheduled_at)) {
+            $nt = new DateTime($this->content->scheduled_at, new DateTimeZone('UTC'));
+            $nt->setTimezone(new DateTimeZone(Yii::$app->settings->get('defaultTimeZone')));
+            $this->content->created_at = $nt->format('Y-m-d H:i:s');
+            $this->content->updated_at = $this->content->created_at;
+          }
+        }
         $this->content->state = $state;
         return true;
     }

```
Of course, this does not implement any options, is not particularly beautiful and changes the feature's default behavior. 

AFAICS it would need a new property (and database flag in the content table), e.g. `update_times_at_schedule` that would be set by the option I suggested, the corresponding update in the `ScheduleOptionsForm` and `scheduleOptions` View,  some adaptation to the `ContentStateService` to update the properties on publish if the corresponding flag is set and translations for the options.

Would that be something you'd consider? I'd be willing to put together a PR (although I'm not quite sure if I can get all the translations correct and how exactly the database migration would work :grimacing: ) if the general approach seems viable. Please let me know what you think.

**Comment by felixhahnweilheim** [comment]
seems related to https://github.com/humhub/humhub/issues/6337

**Comment by ArchBlood** [comment]
Should be fixed by #5315
